tools/autograd/gen_variable_type.py

```diff
@@ -130,21 +130,33 @@
 RECORD_FUNCTION = CodeTemplate("""\
 profiler::RecordFunction profiler("${name}", Function::peek_at_next_sequence_nr());""")
 
+SELECT_OUTPLACE = CodeTemplate("""\
+  if (tracer_state->try_outplace) {
+    ${outplace_trace}
+  } else {
+    ${inplace_trace}
+  }
+""")
+
 PRE_RECORD_TRACE = CodeTemplate("""\
 torch::jit::Node* node = nullptr;
 std::shared_ptr<jit::tracer::TracingState> tracer_state;
 if (jit::tracer::isTracing()) {
   tracer_state = jit::tracer::getTracingState();
-  const static auto op_name = jit::Symbol::fromQualString("aten::${trace_name}");
-  node = tracer_state->graph->create(op_name, /*num_outputs=*/0);
-  jit::tracer::recordSourceLocation(node);
-  ${add_trace_inputs}
-  tracer_state->graph->appendNode(node);
-  ${inplace_guard}
-  jit::tracer::setTracingState(nullptr);
+  ${maybe_select_outplace}
 }
 """)
 
+RECORD_TRACE_BODY = CodeTemplate("""\
+const static auto op_name = jit::Symbol::fromQualString("aten::${trace_name}");
```

Comment:
Do the two functions differ by anything else than the ATen function name? We shouldn't duplicate the whole body in that case.

Reply:
The out of place version has different arguments in some cases as well as a guard to ensure tensor uniqueness.